siva/location.go

```diff
@@ -209,34 +208,39 @@ func (l *Location) setupTransaction(mode borges.Mode) (sivafs.SivaFS, error) {
 		return nil, err
 	}
 
-	l.transactioning = true
+	l.library.startTransaction(l)
+	l.inTransaction = true
```

Comment:
shall we use a mutex to change this safely?

Reply:
Yes. It was implemented without it to test how transactions with sive worked first but this will be done.